shap/explainers/_kernel.py

```diff
@@ -640,12 +640,7 @@ def solve(self, fraction_evaluated, dim):
         nonzero_inds = np.arange(self.M)
         log.debug(f"{fraction_evaluated = }")
         if self.l1_reg == "auto":
-            warnings.warn(
-                "l1_reg='auto' is deprecated and in a future version the behavior will change from a "
-                "conditional use of AIC to simply a fixed number of top features. "
-                "Pass l1_reg='num_features(10)' to opt-in to the new default behaviour.",
-                DeprecationWarning,
-            )
+            warnings.warn("l1_reg='auto' is deprecated and will be removed in a future version.", DeprecationWarning)
```

Comment:
shall we maybe be a bit more specific on which version the removal will happen (I presume it's 0.48)?

Reply:
I'm unsure to be honest. Whilst we should probably aim for 0.48, it's hard to know precisely which version the removal will actually happen as we might desire to make releases for any number of unforeseen reasons. The majority of Deprecation warnings I've seen from other libraries don't include this particular promise, so I'd probably prefer to avoid mentioning a specific future version just to retain flexibility.